hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -244,6 +244,22 @@ public static Schema getRecordKeyPartitionPathSchema() {
     return recordSchema;
   }
 
+  /**
+   * Fetch schema for record key and partition path.
+   */
+  public static Schema getRecordKeyPartitionPathSchema(Schema fileSchema, List<String> recordKeyFields, List<String> partitionPathFields) {
```

Comment:
unit test for this?

Reply:
any reason why we can't just merge the lists outside and keep this method simpler. i.e take a list of fields and get a subschema? in fact, we may have a method like that already, that we can reuse around.

Reply:
looks like we don't have one already. But will fix this method to be generic.